tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/HttpTelemetryTests2.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.TelemetryCore;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.Identity.Test.Common.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Microsoft.Identity.Test.Unit.CoreTests.Telemetry.TelemetryHelperTests;
+using System.Threading;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class HttpTelemetryTests2 : TestBase
+    {
+        private TokenCacheHelper _tokenCacheHelper;
+        private TelemetryManager _telemetryManager;
+        private _TestReceiver _testReceiver;
+        internal _TestEvent _trackingEvent;
+        private const string Comma = ",";
+
+        [TestInitialize]
+        public override void TestInitialize()
+        {
+            base.TestInitialize();
+            _tokenCacheHelper = new TokenCacheHelper();
+            var serviceBundle = TestCommon.CreateServiceBundleWithCustomHttpManager(null, clientId: TestConstants.ClientId);
+            _testReceiver = new _TestReceiver();
+            _telemetryManager = new TelemetryManager(
+                serviceBundle.Config,
+                serviceBundle.PlatformProxy,
+                _testReceiver.HandleTelemetryEvents);
+            _trackingEvent = new _TestEvent("tracking event", "thetelemetrycorrelationid");
+        }
+
+        [TestMethod]
+        public void FetchAndResetPreviousHttpTelemetryContent_ContainsNoStoppedEvents_ReturnsEmptyString()
+        {
+            Assert.AreEqual(string.Empty, _telemetryManager.FetchAndResetPreviousHttpTelemetryContent(_trackingEvent));
+            Assert.AreEqual(0, _telemetryManager.SuccessfulSilentCallCount);
+        }
+
+        [TestMethod]
+        public void FetchCurrentHttpTelemetryContent_ContainsNoEventsInProgress_ReturnsEmptyString()
+        {
+            Assert.AreEqual(string.Empty, _telemetryManager.FetchCurrentHttpTelemetryContent(_trackingEvent));
+        }
+
+        [TestMethod]
+        public async Task AcquireToken_CreateHttpTelemetryHeaders_ReturnsSomethingInterestingAsync()
```

Comment:
Can we have a more complex acceptance test ? i can help add this in - but we need to make sure that: 

- msal client exceptions are captured
- msal service exceptions are captured

My proposed acceptance test is: 

```
1.	Acquire Token Interactive successfully

Current_request = 1 | ATI_ID, 0 | 
Last_request = NULL

2.	Acquire token silent with AT served from cache
3.	Repeat step 2
4.	Acquire Token silent with force_refresh = true -> error invalid_client

Current_request = 1 | ATS_ID, 0 | 
Last_request = NULL

5.	Acquire Token silent with force_refresh = true -> error interaction_required

Current_request = 1 | ATI_ID, 0 | 
Last_request = 1 | 2 | ATS_ID, corr_step_4 | invalid_client


6.	Acquire Token interactive -> error user_cancelled (i.e. no calls to /token endpoint)
7.	Acquire Token interactive -> successful

Current_request = 1 | ATI_ID, 0 |
Last_request = 1 | 2 | | ATS_ID, corr_step_5, ATI_ID, corr_step_6 | interaction_required, user_cancelled | 

1.	Acquire Token Silent with force-refresh -> successful

Current_request = 1 | ATI_ID, 1 |
Last_request = NULL

```